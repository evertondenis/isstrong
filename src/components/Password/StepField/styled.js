import styled from 'styled-components'

export const Field = styled.p`
  margin: 8px 0;
  display: flex;
  align-items: center;

  .icon,
  .isDisabled,
  .isInvalid,
  .isValid {
    margin-right: 8px;
    font-size: 0.75em;
    transition: all .2s linear;
  }

  .isDisabled {
    color: #eaeaf4;
  }

  .isInvalid {
    font-size: 0.75em;
    color: #f79682;
  }

  .isValid {
    font-size: 0.75em;
    color: #1fe6a8;
  }
`
