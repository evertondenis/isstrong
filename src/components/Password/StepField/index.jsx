import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Field } from './styled'

const Circle = ({ className, color }) => (
  <svg className={className} width="10" height="10" viewBox="0 0 10 10" fill="none" xmlns="http://www.w3.org/2000/svg">
    <circle cx="5" cy="5" r="5" fill={color}/>
  </svg>
)

class StepField extends PureComponent {
  render() {
    const { value, children } = this.props
    const status = {
      valid: '#1fe6a8',
      invalid : '#f79682'
    }[value] || '#eaeaf4'

    return (
      <Field>
        <Circle className="icon" color={status} />
        {children}
      </Field>
    )
  }
}

StepField.defaultProps = {
  children: ''
}

StepField.propTypes = {
  children: PropTypes.node,
  value: PropTypes.string.isRequired
}

export default StepField
