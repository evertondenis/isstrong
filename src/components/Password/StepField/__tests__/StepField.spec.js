import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import StepField from '../'
import 'setupTest'

describe('Step Field.', () => {
  it('Should be render <Field />.', () => {
    const pwdLength = 'empty'
    const min = 2
    const wrapper = shallow(
      <StepField value={pwdLength} > Pelo menos {min} caracteres</StepField>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
