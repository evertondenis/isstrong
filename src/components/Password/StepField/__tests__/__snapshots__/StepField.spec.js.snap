// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step Field. Should be render <Field />. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StepField
    value="empty"
>
     Pelo menos 
    2
     caracteres
</StepField>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Circle
          className="icon"
          color="#eaeaf4"
/>,
        Array [
          " Pelo menos ",
          2,
          " caracteres",
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "icon",
          "color": "#eaeaf4",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " Pelo menos ",
      2,
      " caracteres",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Circle
            className="icon"
            color="#eaeaf4"
/>,
          Array [
            " Pelo menos ",
            2,
            " caracteres",
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "icon",
            "color": "#eaeaf4",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " Pelo menos ",
        2,
        " caracteres",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
